src/nbl/video/CVulkanLogicalDevice.h

```diff
@@ -467,6 +465,172 @@ class CVulkanLogicalDevice final : public ILogicalDevice
+                        vk_bufferInfos[bufferInfoOffset + j].buffer = static_cast<const CVulkanBuffer*>(pDescriptorWrites[i].info[j].desc.get())->getInternalObject();
+                        vk_bufferInfos[bufferInfoOffset + j].offset = pDescriptorWrites[i].info[j].info.buffer.offset;
+                        vk_bufferInfos[bufferInfoOffset + j].range = pDescriptorWrites[i].info[j].info.buffer.size;
+                    }
+                    else
+                    {
+                        vk_bufferInfos[bufferInfoOffset + j] = dummyInfo;
+                    }
+                }
+
+                vk_writeDescriptorSets[i].pBufferInfo = vk_bufferInfos.data() + bufferInfoOffset;
+                bufferInfoOffset += pDescriptorWrites[i].count;
+            } break;
+
+            case asset::IDescriptor::EC_IMAGE:
+            {
+                const auto& firstDescWriteImageInfo = pDescriptorWrites[i].info[0].info.image;
+
+                VkDescriptorImageInfo dummyInfo = { VK_NULL_HANDLE };
+                if (firstDescWriteImageInfo.sampler && (firstDescWriteImageInfo.sampler->getAPIType() == EAT_VULKAN))
+                    dummyInfo.sampler = static_cast<const CVulkanSampler*>(firstDescWriteImageInfo.sampler.get())->getInternalObject();
+                dummyInfo.imageView = static_cast<const CVulkanImageView*>(pDescriptorWrites[i].info[0].desc.get())->getInternalObject();
+                dummyInfo.imageLayout = static_cast<VkImageLayout>(pDescriptorWrites[i].info[0].info.image.imageLayout);
+
+                for (uint32_t j = 0u; j < pDescriptorWrites[i].count; ++j)
+                {
+                    const auto& descriptorWriteImageInfo = pDescriptorWrites[i].info[j].info.image;
+                    if (descriptorWriteImageInfo.imageLayout != asset::IImage::EL_UNDEFINED)
+                    {
+                        VkSampler vk_sampler = VK_NULL_HANDLE;
+                        if (descriptorWriteImageInfo.sampler && (descriptorWriteImageInfo.sampler->getAPIType() == EAT_VULKAN))
+                            vk_sampler = static_cast<const CVulkanSampler*>(descriptorWriteImageInfo.sampler.get())->getInternalObject();
+
+                        VkImageView vk_imageView = static_cast<const CVulkanImageView*>(pDescriptorWrites[i].info[j].desc.get())->getInternalObject();
+
+                        vk_imageInfos[imageInfoOffset + j].sampler = vk_sampler;
+                        vk_imageInfos[imageInfoOffset + j].imageView = vk_imageView;
+                        vk_imageInfos[imageInfoOffset + j].imageLayout = static_cast<VkImageLayout>(descriptorWriteImageInfo.imageLayout);
+                    }
+                    else
+                    {
+                        vk_imageInfos[imageInfoOffset + j] = dummyInfo;
+                    }
+                }
+
+                vk_writeDescriptorSets[i].pImageInfo = vk_imageInfos.data() + imageInfoOffset;
+                imageInfoOffset += pDescriptorWrites[i].count;
```

Comment:
here's a trick:
1. `push_back` onto `vk_imageInfos`
2. set `pImageInfo` to `reinterpret_cast<VkDescriptorImageInfo*>(nullptr)+vk_imageInfos.size()` **before the `for` loop which `push_back`s onto the `vk_imageInfos`**
3. after you get our of the `for` which ecapsulates the `switch` patch up the pointers to take `vk_imageInfos.data()` into account

then

Reply:
TODO

Reply:
> so I'm suggesting only store the offset casted to a pointer

It makes a a lot of sense when you put it like this.

I just want to make one small modification: since we're dealing with offsets into the `vector`, we don't need the `reinterpret_cast<VkDescriptorImageInfo*>(nullptr)` because, according to the rules of pointer arithmetic, that will give us the byte offset.

Reply:
Resolved.